implementations/rust/ockam/ockam_command/src/enroll/oidc_service.rs

```diff
@@ -61,14 +66,29 @@ impl OidcServiceExt for OidcService {
         }
         // Otherwise, write the instructions at stderr as normal
         else {
+            clipboard = Clipboard::new();
+            let otc_string = clipboard
+                .as_mut()
+                .ok()
+                .and_then(|clip| clip.set_text(dc.user_code.to_string()).ok())
```

Comment:
I'm converting to `Option` here because otherwise there would be an `arboard::Error` that can't be converted to the type that `as_mut()` needs it to be (`&mut arboard::Error`).

This problem stemmed from needing to keep the clipboard in scope for the length of the interaction. This was definitely annoying and I wasn't able to figure out a better, simpler way of doing it. There is some documentation for setting up an `arboard` "daemon" that forks and runs in the background, which would allow other invocations to the clipboard to immediately drop from memory, but that seemed like a lot more effort for just a little bit simpler code.

Reply:
I think your implementation is perfectly fine, given the expected lifetime of the clipboard! 👍